mhkit/wave/contours.py

```diff
@@ -1521,4 +1522,331 @@ def _bivariate_KDE(x1, x2, bw, fit, nb_steps, Ndata_bivariate_KDE, kwargs):
+    user-defined T values.
+
+    Parameters
+    ----------
+    t_samples : nparray
+        Points for sampling along return contour
+    t_contour : np.array
+        T values along contour
+    hs_contour : np.array
+        Hs values along contour
+
+    Returns
+    -------
+    hs_samples : nparray
+        points sampled along return contour
+    """
+    #finds minimum and maximum energy period values
+    amin = np.argmin(t_contour)
+    amax = np.argmax(t_contour)
+    aamin = np.min([amin, amax])
+    aamax = np.max([amin, amax])
+    #finds points along the contour
+    w1 = hs_contour[aamin:aamax]
+    w2 = np.concatenate((hs_contour[aamax:], hs_contour[:aamin]))
+    if (np.max(w1) > np.max(w2)):
+        x1 = t_contour[aamin:aamax]
+        y = hs_contour[aamin:aamax]
+    else:
+        x1 = np.concatenate((t_contour[aamax:], t_contour[:aamin]))
+        y1 = np.concatenate((hs_contour[aamax:], hs_contour[:aamin]))
+    #sorts data based on the max and min energy period values
+    ms = np.argsort(x1)
+    x = x1[ms]
+    y = y1[ms]
+    #interpolates the sorted data
+    si = interp.interp1d(x, y)
+    #finds the wave height based on the user specified energy period values
+    hs_samples = si(t_samples)
+
+    return hs_samples
+
+
+def _generate_sample_data(beta_lines, rho_zeroline, theta_zeroline,
+                          points_per_interval, contour_probs):
+    """ Calculate radius, angle, and weight for each sample point
+
+    Parameters
+    ----------
+    beta_lines: np.array
+        Array of mu fitting function parameters.
+    rho_zeroline: np.array
+        Array of radii
+    theta_zeroline: np.array
+    points_per_interval: np.array
+    contour_probs: np.array
+
+    Returns
+    -------
+    sample_alpha: np.array
+        Array of fitted sample angle values.
+    sample_beta: np.array
+        Array of fitted sample radius values.
+    weight_points: np.array
+        Array of weights for each point.
+    """
+    num_samples = (len(beta_lines) - 1) * points_per_interval
+    alpha_bounds = np.zeros((len(beta_lines) - 1, 2))
+    angular_dist = np.zeros(len(beta_lines) - 1)
+    angular_ratio = np.zeros(len(beta_lines) - 1)
+    alpha = np.zeros((len(beta_lines) - 1, points_per_interval + 1))
+    weight = np.zeros(len(beta_lines) - 1)
+    sample_beta = np.zeros(num_samples)
+    sample_alpha = np.zeros(num_samples)
+    weight_points = np.zeros(num_samples)
+
+    # Loop over contour intervals
+    for i in range(len(beta_lines) - 1):
+        # Check if any of the radii for the Hs=0, line are smaller than
+        # the radii of the contour, meaning that these lines intersect
+        # r = rho_zeroline - beta_lines[i + 1]  # TODO
+        r = rho_zeroline - beta_lines[i + 1] + 0.01
+        if any(r < 0):
+            # left = np.amin(np.where(r < -0.01))  # TODO
+            # right = np.amax(np.where(r < -0.01)) # TODO
+            # TODO
```

Comment:
Why is there a TODO here? Do we still need to do this?

Reply:
This was resolved. Need to delete the TODOs and the commented out lines.

Reply:
Done w/ @hivanov-nrel latest commit.